docs/DynamicPropertyIteration.md

```diff
@@ -0,0 +1,309 @@
+# Dynamic property iteration using key paths
+
+[Richard Wei](https://github.com/rxwei), [Dan Zheng](https://github.com/dan-zheng)
+
+Last updated: March 2019
+
+# Background and motivation
```

Comment:
This should be double-#. These are headings, not titles.

Reply:
Good catch! Fixed in https://github.com/tensorflow/swift/pull/132/commits/09f5f121db353224d576fb5b9aa5d8a26a6a06fd.